pydropsonde/pipeline.py

```diff
@@ -306,13 +308,144 @@ def iterate_method_over_dataset(
     return result
 
 
-def gridded_to_pattern(
+def get_circle_times_from_yaml(config: configparser.ConfigParser):
+
+    yaml_directory = config.get("MANDATORY", "segments_directory")
+    allyamlfiles = sorted(glob.glob(yaml_directory + "*.yaml"))
+
+    circle_times = []
+    sonde_ids = []
+    flight_date = []
+    platform_name = []
+    segment_id = []
+
+    for i in allyamlfiles:
+        with open(i) as source:
+            flightinfo = yaml.load(source, Loader=yaml.SafeLoader)
+
+        circle_times.append(
+            [
+                (c["start"], c["end"])
+                for c in flightinfo["segments"]
+                if "circle" in c["kinds"]
+                if len(c["dropsondes"]["GOOD"]) >= 6
+            ]
+        )
+
+        sonde_ids.append(
+            [
+                c["dropsondes"]["GOOD"]
+                for c in flightinfo["segments"]
+                if "circle" in c["kinds"]
+                if len(c["dropsondes"]["GOOD"]) >= 6
+            ]
+        )
+
+        segment_id.append(
+            [
+                (c["segment_id"])
+                for c in flightinfo["segments"]
+                if "circle" in c["kinds"]
+                if len(c["dropsondes"]["GOOD"]) >= 6
+            ]
+        )
+
+        if "HALO" in i:
+            platform_name.append("HALO")
+        elif "P3" in i:
+            platform_name.append("P3")
+        else:
+            platform_name.append("")
```

Comment:
I would strongly discourage to extract information from the file name. File names may change, or may not be present at all (e.g. for the flight segmentation from eurec4a, the recommended way to use flight segmentation information through the all_flights.yaml file which is available through a compiled release).
Instead, you could just use the platform identifier which is given within the yaml file.

```suggestion
        platform_name.append(flightinfo["platform"]
```